### PR TITLE
Windows Separation Remastered: Welcome

### PR DESCRIPTION
**Pull request by lolgear**
- [x] Welcome window has been incapsulated.
- [x] Welcome window has been moved to its own xib.
- [x] Welcome window has been removed from MainMenu xib.
- [x] App Delegate loading from nibs has been added.
- [x] App Delegate windows post setup has been added.

### CODE REVIEW

**Comment by lolgear** [comment]
@lucasderraugh 
It would be pleasant, if we will do it in a row, in this case I don't need to move master to actualize project and main menu xib if some PRs change them :)

**Comment by lucasderraugh** [comment]
Ok, I'll try to make them a priority.

File: GitUp/Application/AppDelegate.m [hunk 1]
```diff
@@ -189,46 +191,10 @@ - (void)_openDocument:(NSMenuItem*)sender {
...
-    }
-  } else {
-    NSMenuItem* item = [[NSMenuItem alloc] initWithTitle:NSLocalizedString(@"No Repositories", nil) action:NULL keyEquivalent:@""];
-    item.enabled = NO;
-    [menu addItem:item];
-  }
-}
-
 - (void)awakeFromNib {
-  _welcomeMaxHeight = _welcomeWindow.frame.size.height;
```
Comment (lucasderraugh) [inline GitUp/Application/AppDelegate.m]: Good cleanup 😄

File: GitUp/Application/AppDelegate.m [hunk 2]
```diff
@@ -299,6 +263,36 @@ - (void)handleDocumentCountChanged {
...
+#pragma mark - Loading Xibs
+- (id)_loadWindowFromBundleXibWithName:(NSString *)name expectedClass:(Class)class {
+  NSBundle *mainBundle = [NSBundle mainBundle];
+  NSArray *objects = @[];
+  [mainBundle loadNibNamed:name owner:self topLevelObjects:&objects];
+  NSArray *filteredObjects = [objects filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(id  _Nullable evaluatedObject, NSDictionary<NSString *,id> * _Nullable bindings) {
+    return [evaluatedObject isKindOfClass:class];
+  }]];
+  return filteredObjects.firstObject;
+}
```
Comment (lucasderraugh) [inline GitUp/Application/AppDelegate.m]: Do we expect to use this for other classes. My ideal would be that we just have a WelcomeWindow init method that does all the xib loading work for us. Could we perhaps move this logic into that class? Other than that, I think this PR looks pretty good.

Comment (lolgear) [reply GitUp/Application/AppDelegate.m]: @lucasderraugh 
In previous PR I load windows and About panel by this method.
Also, look at `-loadNibNamed` invocation. It specifies owner as `self` ( AppDelegate ). Other windows don't have too much incapsulated logic ( I leave them `as-is` with some methods in `AppDelegate` ).

**Comment by lolgear** [comment]
@lucasderraugh Could you merge it?

**Pull request merged**